File: src/main/java/de/silencio/activecraftcore/messages/Reasons.java
package de.silencio.activecraftcore.messages;

public class Reasons extends ActiveCraftCoreMessage {


    public static String HACKING() {
        return getMessage(MessageType.REASON, "hacking");
    }

    public static String BOTTING() {
        return getMessage(MessageType.REASON, "botting");
    }

    public static String UNAUTHORIZED_ALTERNATE_ACCOUNT() {
        return getMessage(MessageType.REASON, "unauthorized-alt-account");
    }

    public static String SPAM() {
        return getMessage(MessageType.REASON, "spam");
    }

    public static String ABUSIVE_LANGUAGE() {
        return getMessage(MessageType.REASON, "abusive-language");
    }

    public static String STEALING() {
        return getMessage(MessageType.REASON, "stealing");
    }

    public static String GRIEFING() {
        return getMessage(MessageType.REASON, "griefing");
    }

    public static String MODERATOR_BANNED() {
        return getMessage(MessageType.REASON, "moderator-banned");
    }

    public static String MODERATOR_WARNED() {
        return getMessage(MessageType.REASON, "moderator-warned");
    }

    public static String MODERATOR_KICKED() {
        return getMessage(MessageType.REASON, "moderator-kicked");
    }
}

File: src/main/java/de/cplaiz/activecraftcore/messagesv2/messagecollections/Reasons.kt
package de.cplaiz.activecraftcore.messagesv2.messagecollections

import de.cplaiz.activecraftcore.messagesv2.MessageSupplier

class Reasons(val messageSupplier: MessageSupplier) {

    val hacking: String
        @JvmName("hacking") get() = messageSupplier.getRawMessage("reason.hacking")

    val botting: String
        @JvmName("botting") get() = messageSupplier.getRawMessage("reason.botting")

    val unauthorizedAlternateAccount: String
        @JvmName("unauthorizedAlternateAccount") get() = messageSupplier.getRawMessage("reason.unauthorized-alt-account")

    val spam: String
        @JvmName("spam") get() = messageSupplier.getRawMessage("reason.spam")

    val abusiveLanguage: String
        @JvmName("abusiveLanguage") get() = messageSupplier.getRawMessage("reason.abusive-language")

    val stealing: String
        @JvmName("stealing") get() = messageSupplier.getRawMessage("reason.stealing")

    val griefing: String
        @JvmName("griefing") get() = messageSupplier.getRawMessage("reason.griefing")

    val moderatorBanned: String
        @JvmName("moderatorBanned") get() = messageSupplier.getRawMessage("reason.moderator-banned")

    val moderatorWarned: String
        @JvmName("moderatorWarned") get() = messageSupplier.getRawMessage("reason.moderator-warned")

    val moderatorKicked: String
        @JvmName("moderatorKicked") get() = messageSupplier.getRawMessage("reason.moderator-kicked")
}